### PR TITLE
Exit game options

### PR DESCRIPTION
**Pull request by lpenzey**
Added Escape module to include options for quitting/restarting throughout gameplay

### CODE REVIEW

File: Lib/mastermind.rb [hunk 1]
```diff
@@ -1,5 +1,25 @@
...
+
+  def restart
+    puts "Would you like to quit or restart? Type \"q\" to quit or \"r\" to restart."
+        if gets.chomp.downcase == "r"
+          load 'mastermind.rb' 
+        else
+          abort("Thanks for playing!")
+        end
+      end
+end
```
Comment (akiraandy) [inline Lib/mastermind.rb]: I would rename this module to something like "Commands"

Comment (akiraandy) [reply Lib/mastermind.rb]: On the lines where you `load 'mastermind.rb` I would extract those to a constant so if your path ever changes for `mastermind.rb` you only have to change it in one place.

File: Lib/mastermind.rb [hunk 2]
```diff
@@ -82,6 +99,7 @@ def initialize(new_input)
   def get_input
     puts "Please enter a guess [****]"
     @player_input = gets.chomp.downcase.gsub(/[\W]/, "")
+    leave if @player_input == "quit"
```
Comment (akiraandy) [inline Lib/mastermind.rb]: I like how this reads like English 👍

File: Spec/mastermind_spec.rb [hunk 3]
```diff
@@ -101,4 +101,17 @@
...
+
+  it "returns invalid length message" do 
+    expect { response.length_error }.to output(/Invalid length/).to_stdout
+  end
+
+  it "quits the game if \"quit\" is entered" do
+    response = Response.new("quit")
+    expect{ response.leave }.to output(/Are you sure you want to quit?/).to_stdout
+  end
 end
```
Comment (akiraandy) [inline Spec/mastermind_spec.rb]: These last two tests are pretty brittle. I wouldn't recommend testing output to the console as a reliable test of behavior. I would even recommend just cutting these last two tests out. `leave` and `restart` aren't doing a lot except for loading and exiting the application which we know Ruby knows how to do with `load` and `abort`. Maybe this is an example of over-testing.

File: Lib/mastermind.rb (hunk 2 above)
Comment (lpenzey) [reply Lib/mastermind.rb]: thanks!

**Pull request merged**